pennylane/tape/tape.py

```diff
@@ -210,6 +210,7 @@ def expand_tape(tape, depth=1, stop_at=None, expand_measurements=False):
 
     # Update circuit info
     new_tape._update_circuit_info()
+    new_tape._output_dim = tape.output_dim
```

Comment:
We should include this as an improvement for the Operator refactor --- all measurements should have their own `shape` property, that indicates the array shape they return.

E.g.,

```python
>>> qml.expval(qml.PauliZ(0)).shape
(1,)
>>> qml.probs(wires=[0, 1, 2]).shape
(2, 2, 2)
```

This way, we can compute the output shape of a tape _in advance_ to any evaluation, and with total accuracy!

Reply:
Tagging @mariaschuld for visibility once back from vacation 🙂

Reply:
This is a great idea and would be super useful when working with JAX. :slightly_smiling_face: